File: batch.js
// calc num queries
total = 9999
page_size = 100
num_queries = Math.ceil(total/page_size)
console.log('num_queries: ' + num_queries)

// calc pages
pages = []
for (let i = 0; i < num_queries; i++) {
    pages.push(i*page_size)
}
console.log(pages)

// determine batch size
batch_size = 2
console.log('batch_size: ' + batch_size)

for (let n = 0; n < num_queries; n++) {
    slice = pages.slice(n * batch_size, (n + 1) * batch_size)
    if (slice.length > 0){
        console.log(n+'_slice: ' + slice)
    } else {
        break;
    }
}

File: batch.py
import math

# calc num queries
total = 9999
page_size = 100
num_queries = int(math.ceil(float(total) / float(page_size)))

print(num_queries)
print('num_queries: ' + str(num_queries))

# calc pages
pages = []
for i in range(num_queries):
    pages.append(i*page_size)
print(pages)

# determine batch size
batch_size = 40
print('batch_size: ' + str(batch_size))

for n in range(num_queries):
    slice = pages[n * int(batch_size): (n + 1) * int(batch_size)]
    if len(slice) > 0:
        print(str(n)+'_slice: ' + str(slice))
    else:
        break
